pkg/image/api/validation/validation_test.go

```diff
@@ -759,6 +761,20 @@ func TestValidateImageStreamImport(t *testing.T) {
 				field.Invalid(field.NewPath("spec", "repository", "from", "name"), "a/b@sha256:something", "invalid reference format"),
 			},
 		},
+		"registry not allowed for image import": {
+			isi: &api.ImageStreamImport{ObjectMeta: validMeta, Spec: repoFn("unknown.registry:5000/foo/bar")},
+			expected: field.ErrorList{
+				field.Invalid(field.NewPath("spec", "repository", "from", "name"), "unknown.registry:5000/foo/bar", `importing images from registry "unknown.registry" is forbidden, only images from "foo" are allowed`),
+			},
+			allowedRegistries: &configapi.AllowedRegistries{
+				{DomainName: "foo"},
+			},
+		},
+		"importing from unknown registry allowed when allowed registries is not set (legacy)": {
+			isi:               &api.ImageStreamImport{ObjectMeta: validMeta, Spec: repoFn("unknown.registry:5000/foo/bar")},
+			expected:          field.ErrorList{},
+			allowedRegistries: &configapi.AllowedRegistries{},
+		},
```

Comment:
Can you add one more test case with `AllowedRegistries` not empty and matching the registry?